seed/analysis_pipelines/pipeline.py

```diff
@@ -156,28 +160,78 @@ def _stop_task_chain(task_instance):
                 _stop_task_chain(_self)
                 return
             except Exception as e:
+                # yikes, we didn't account for this. Try to bow out as gracefully as possible
+                reraise_exception = False
                 try:
-                    Analysis.objects.get(id=_analysis_id)
-
+                    with transaction.atomic():
+                        # don't wait for lock to avoid deadlock
+                        locked_analysis = (
+                            Analysis.objects
+                            .select_for_update(nowait=True)
+                            .get(id=_analysis_id)
+                        )
+
+                        # analysis still exists, and we were able to get a lock on it
+                        # something unexpected happened during the task
+                        if locked_analysis.in_terminal_state():
```

Comment:
a new case I added in this commit. If something goes bad, and someone already put the analysis in a terminal state, just log and exit (don't raise an exception or create an analysis message b/c we could have already ended up in some invalid state)